File: cacao/ui/components/data/avatar/avatar.js
(component) => {
    console.log("[CacaoCore] Rendering avatar component:", component);
    const el = document.createElement("span");
    el.className = "avatar";
    
    if (component.props.shape) {
        el.classList.add(component.props.shape);
    }
    if (component.props.size) {
        el.classList.add(component.props.size);
    }

    if (component.props.src) {
        const img = document.createElement("img");
        img.src = component.props.src;
        img.alt = "Avatar";
        el.appendChild(img);
    } else if (component.props.icon) {
        const icon = document.createElement("span");
        icon.className = "avatar-icon";
        applyContent(icon, component.props.icon);
        el.appendChild(icon);
    }

    return el;
}

File: cacao/ui/components/data/avatar/avatar.py
"""
Avatar component for representing a user or entity with an image or icon.
"""

from typing import Any, Dict
from ...base import Component


class Avatar(Component):
    """
    Represents a user or entity with an image or icon.
    """
    def __init__(
        self,
        src = None,
        icon = None,
        shape = "circle",
        size = "default",
        **kwargs
    ) -> None:
        self.src = src
        self.icon = icon
        self.shape = shape
        self.size = size
        self.extra_props = kwargs

    def render(self) -> Dict[str, Any]:
        return {
            "type": "avatar",
            "props": {
                "src": self.src,
                "icon": self.icon,
                "shape": self.shape,
                "size": self.size,
                **self.extra_props
            }
        }